semgrep-core/src/runner/Run_semgrep.ml

```diff
@@ -432,9 +432,15 @@ let mk_rule_table rules list_of_rule_ids =
   let rule_pairs = Common.map (fun r -> (fst r.R.id, r)) rules in
   let rule_table = Common.hash_of_list rule_pairs in
   let id_pairs =
-    List.mapi
-      (fun i rule_id -> (i, Hashtbl.find rule_table rule_id))
-      list_of_rule_ids
+    list_of_rule_ids
+    |> List.mapi (fun i x -> (i, x))
+    (* We filter out rules here if they don't exist, because we might have a
+     * rule_id for an extract mode rule, but extract mode rules won't appear in
+     * rule pairs, because they won't be in the table we make for search
+     * because we don't want to run them at this stage.
```

Comment:
Let's make the assumption explicit (all search/taint rule ids are in this table). There might be another comment related to the rule_table which needs to be updated?

Reply:
> Let's make the assumption explicit (all search/taint rule ids are in this table).

In terms of just documenting this for the usage in https://github.com/returntocorp/semgrep/blob/58cac3cff1c0989582fceab63014f3e63c086e0b/semgrep-core/src/runner/Run_semgrep.ml#L600